sdlib/d/gc/bin.d

```diff
@@ -153,3 +164,64 @@ private:
 		return current;
 	}
 }
+
+unittest batchAllocation {
+	import d.gc.util;
+	shared Arena arena;
+
+	auto base = &arena.base;
+	scope(exit) arena.base.clear();
+
+	import d.gc.emap;
+	static shared ExtentMap emap;
+	emap.tree.base = base;
+
+	import d.gc.region;
+	shared RegionAllocator regionAllocator;
+	regionAllocator.base = base;
+
+	arena.regionAllocator = &regionAllocator;
+
+	void*[2048] buffer;
+
+	// Get and verify 4 slabs' worth of allocs for each small size class
+	auto expectedSlabs = 4;
+
+	import d.gc.sizeclass;
+	foreach (sc; 0 .. ClassCount.Small) {
```

Comment:
On the other hand, checking all size class isn't really necessary. Checking a couple, with a couple of slot count could be useful.